test/instrumentation/modules/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,196 @@
+'use strict'
+
+process.env.ELASTIC_APM_TEST = true
+const host = (process.env.ES_HOST || 'localhost') + ':9200'
+const node = 'http://' + host
+
+const agent = require('../../../..').start({
+  serviceName: 'test',
+  secretToken: 'test',
+  captureExceptions: false,
+  metricsInterval: 0,
+  centralConfig: false
+})
+
+const test = require('tape')
+
+const { Client } = require('@elastic/elasticsearch')
+
+const mockClient = require('../../../_mock_http_client')
+const findObjInArray = require('../../../_utils').findObjInArray
+
+test('client.ping with promise', function userLandCode (t) {
+  resetAgent(done(t, 'HEAD', '/'))
+
+  agent.startTransaction('foo')
+
+  const client = new Client({ node })
+
+  client.ping().then(function () {
+    agent.endTransaction()
+    agent.flush()
+  }).catch(t.error)
+})
+
+test('client.ping with callback', function userLandCode (t) {
+  resetAgent(done(t, 'HEAD', '/'))
+
+  agent.startTransaction('foo')
+
+  const client = new Client({ node })
+
+  client.ping(function (err, result) {
+    t.error(err)
+    agent.endTransaction()
+    agent.flush()
+  })
+})
+
+test('client.search with callback', function userLandCode (t) {
+  resetAgent(done(t, 'GET', '/_search', 'q=pants'))
```

Comment:
In the old client we actually report this as `{"q":"pants"}` because that's what the user provides. The fact that the client chooses the add this to the querystring is not really relevant. It would be nicest if we reported the query as it was provided by the user.

The downside of that is that we'll have to serialize it twice, so the question is if it's worth it.